### DISCUSSION TITLE
What should be checked-in to github?

### DISCUSSION THREAD

**Discussion by venkyvb** [goreleaser/goreleaser#1666] [Q&A] [answered]
I am trying out goreleaser for a new go-project that I am working on. I ran the `goreleaser init` which generated the `.goreleaser.yml` file.

Then I ran `goreleaser --snapshot` which generated the `dist` folder and the associated artifacts.

However when I try to run `goreleaser` I see the following error -

```
   • releasing...     
   • loading config file       file=.goreleaser.yml
   • running before hooks
      • running go mod download
      • running go generate ./...
   • loading environment variables
   • getting and validating git state
      • releasing v0.1.0, commit 26759656b03c1516883397d5cff0ef4b584cd779
   ⨯ release failed after 0.22s error=git is currently in a dirty state, please check in your pipeline what can be changing the following files:
?? .goreleaser.yml
?? dist/
```
Am I missing a step here (should I be committing the `.goreleaser.yml` and the `dist` folder before doing the `goreleaser` command?

**Comment by radeksimko** [comment, marked as answer]
Generally `dist` is only a (temporary) folder where GoReleaser stores artifacts such as binaries or archives, before publishing them e.g. to GitHub via the API. As such `dist` folder would typically _not_ be checked-in and you may even go as far as putting it into `.gitignore`.

`.goreleaser.yml` on the other hand is the config, which does need to be checked-in.

> should I be committing the .goreleaser.yml and the dist folder before doing the goreleaser command?

Yes, the error you pasted above is related to the fact that you made changes in your config, but haven't yet committed them (hence left git in "dirty state") which in turn makes it impossible for git (an in turn GoReleaser) to uniquely identify the build that it would produce - there is no commit SHA. That is why even for snapshots everything has to be committed as snapshots are identified by SHA (unlike full builds which are identified by tags).

**Reply by venkyvb** [reply]
Thanks @radeksimko. This helps. Wondering if we should add this to the documentation? (about checking in only `.goreleaser.yml`)

**Reply by radeksimko** [reply]
I am not sure where exactly would this be best placed in the documentation, but PRs are welcomed. 😉

**Reply by venkyvb** [reply]
Have added this to the quick-start.md. Also opened a PR for the same (my first PR to an open source project in general 😃) - https://github.com/goreleaser/goreleaser/pull/1667
